source/parse.h

```diff
@@ -520,6 +520,10 @@ struct type_id_node
     source_position pos;
 
     std::vector<token const*> pc_qualifiers;
+    token const* address_of                 = {};
+    token const* dereference_of             = {};
+    int dereference_cnt                     = {};
+    token const* suspicious_initialization  = {};
```

Comment:
This additional data also is localized. Have you checked to make sure there are no lifetime issues (e.g., we always point to parents that will be destroyed after we are, or else don't use the pointers after the targets might be destroyed)?

Reply:
I have not checked that. I assumed that the lifetime will be the same as `pc_qualifiers` - need to double check that.

Reply:
Ah, I made a thinko. You're fine here for lifetime -- you're pointing to `token`s which outlive everything. For some reason I was thinking you were pointing to grammar nodes, but you aren't -- no worries then, sorry for the distraction.